iec104 north version 1.0.0

aklira opened the pull request:
This is iec104 north version 1.0.0

**Features:**
- Connection management
- Support for multiple redundancy group
- Protocol stack configuration
- Exchanged data configuration
- Support for general interrogation
- GI interrogation in case of empty storage layer
- Support for time synchronization command
- Handle commands feedback (Acknowledgment)
- Handle messages in monitoring direction
- Handle commands in control direction
- Support for direct and select/execute
- TLS support

**Bug fixes:**
- fixed problem forwarding messages with COT=11 or 12 or 2
- forward negative flag in ACT-CON and ACT-TERM messages

MarkRiddoch committed:
FOGL-4713 Make requirements.sh executable

Signed-off-by: Mark Riddoch <mark@dianomic.com>

MarkRiddoch committed:
Merge pull request #1 from fledge-iot/FOGL-4713

FOGL-4713 Make requirements.sh executable

Commit:
fixed problem with ASDU types not recognized

pintomax committed:
FOGL-5821: make plugin config static

FOGL-5821: make plugin config static

pintomax committed:
Merge pull request #2 from fledge-iot/FOGL-5821-make-plugin-config-static

FOGL-5821: make plugin config static

aklira committed:
added mapping to pivot format reading

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Merge pull request #3 from aklira/develop

added mapping to pivot format reading

aklira committed:
Merge pull request #3 from aklira/develop

added mapping to pivot format reading
Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Update CMakeLists.txt

Made CMakeLists consistent with iec104 south plugin
Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Merge branch 'fledge-iot:develop' into develop

Signed-off-by: akli <akli.rahmoun@gmail.com>
Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
added lib60870 include location

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
fixed issue with wrong method for DoublePointWithCP56Time2a

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Merge pull request #5 from aklira/develop

Fixed some issues

aklira committed:
Create qualityscan.yml

aklira committed:
Update qualityscan.yml

aklira committed:
Update qualityscan.yml

aklira committed:
Update qualityscan.yml

aklira committed:
Update .gitignore

ignore .github content

aklira committed:
added unit tests and coverage

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Update README.rst

aklira committed:
Update and rename qualityscan.yml to ci.yml

aklira committed:
Update ci.yml

aklira committed:
Update requirements.sh

aklira committed:
Update ci.yml

mend-bolt-for-github[bot] committed:
Add .whitesource configuration file

aklira committed:
Merge pull request #1 from fledge-power/whitesource/configure

Configure WhiteSource Bolt for GitHub

aklira committed:
added libgmock-dev installation

aklira committed:
initial commit

jeanericdu31 committed:
Tested handlers for messages and starting.

Signed-off-by: jeanericdu31 <jean-eric.nzaramba@ext.soprasteria.com>

aklira committed:
Merge pull request #2 from fledge-power/PMP-140_141

Tested handlers for messages and starting.

aklira committed:
Update ci.yml

mzillgith committed:
- added code to forward spontaneous and periodic messages from south side
- added code to handle interrogation requests from clients

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added support to handle interrogation requests with broadcast address

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- store data points with data type instead of message type

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- use rapidjson for configuration parsering
- refactored configuration code in separate class/module

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- removed old json header files

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added basic implemenation of command handling

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- implemented configuration of commands in control direction

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added support for redundancy groups

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- show IP address of connecting clients in log

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- implement handling of transport_layer configuration parameters

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added handling of application layer configuration parameters

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added handling of time sync commands
- implemented OA filter list for commands

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added support for parameter application_layer.asdu_queue_size to configure ASDU queue size
- added support for parameter application_layer.accept_cmd_with_time to configure usage of time stamps in commands

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- fixed test cases; code cleanup

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added missing test case files

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- fixed error: "object backing the pointer will be destroyed at the end of the full-expression"

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #14 from fledge-power/mz_refactoring

Mz refactoring

aklira committed:
Merge pull request #15 from fledge-power/develop

Develop

mzillgith committed:
- added missing memory cleanup

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- handling of command feedback (ACT-CON, ACT-TERM) from south sode

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- updated operate signature for control direction to be fledge 2.0.0 compatible

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- moved source code files (.cpp)  to separate folder (src)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- fixed handling of select in command feedback

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added "control_target" configuration option to specify target service for commands

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- implemented "cmd_exec_timeout" configuration parameter

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- renamed control_target parameter to cmd_dest

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added additional checks
- updated test configurations

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #21 from fledge-power/mz_refactoring

Mz refactoring

aklira committed:
Update ci.yml

mzillgith committed:
- added support for monitoring types with CP56Time2a time tags

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #22 from fledge-power/mz_refactoring

- added support for monitoring types with CP56Time2a time tags

mzillgith committed:
- fixed problem with time stamp handling of spontaneous messages
- added more test cases

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #24 from fledge-power/mz_refactoring

- fixed problem with time stamp handling of spontaneous messages

mzillgith committed:
- Implemented basic TLS support

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- fixed problem with TLS configuration

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- added test certificates

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- updated TLS test cases to run with CI workflow

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- spontaneously transmit additional COTs (background-scan, remote-info-remote, remote-info-local)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #25 from fledge-power/mz_refactoring

- Implemented basic TLS support

aklira committed:
Update VERSION

mzillgith committed:
- send negative ACT-CON when time in invalid in command with timestamp (#27)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #28 from fledge-power/mz_refactoring

- send negative ACT-CON when time in invalid in command with timestam…

aklira committed:
Update ci.yml

mzillgith committed:
- forward negative flag in ACT-CON and ACT-TERM messages

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #29 from fledge-power/mz_refactoring

- forward negative flag in ACT-CON and ACT-TERM messages

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
Update ci.yml

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

mzillgith committed:
- started to implement north side connection handling based on mode and south side connection status (RFG-70)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- handle control center connection based on mode and south connection status (#31)(RFD-70)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- send request_connection_status operation after startup (RFD-70)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- send request_connection_status command with parameter

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
update to version 1.0.0

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

mzillgith committed:
- fixed control test cases

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
Merge branch 'main' into mz_refactoring

mzillgith committed:
- update ci.yml

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- updated ci.yml

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Merge pull request #33 from fledge-power/mz_refactoring

Mz refactoring

MarkRiddoch reviewed (dismissed):
Hi Akli, you need to sign your commits before we can merge them in. It is a requirement of the Linux Foundation that all commits are signed.

ashish-jabble commented:
Also, Please create a Pull request against develop branch. main branch is only for the released version.(i.e last/latest)

aklira committed:
Update issue templates

mzillgith committed:
- fixed - random IV bit and date when disconnecting RTU (RFD-79)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- fixed - sending empty ASDU when no data points are configured for interroagion (RFD-80)

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

aklira committed:
Update publish.yml

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
Merge pull request #38 from fledge-power/mz_refactoring

Mz refactoring

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
changed south event asset name

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
changed south event dp name

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Merge pull request #40 from fledge-power/39-south-event

changed south event asset name

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

jude-adam committed:
Added support for gi groups

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

Commit:
Added tests

Signed-off-by: Maurício <mauricio.fernandes@gridcorelabs.com>

jude-adam committed:
Changed command format

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

jude-adam committed:
Fixed pem certificate search folder

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

mzillgith committed:
Merge branch 'RFD-85' into mz_refactoring

mzillgith committed:
Merge branch 'RFD-89' into mz_refactoring

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

mzillgith committed:
- removed includes for gmock

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

Commit:
Merge branch 'mz_refactoring' of bitbucket.org:mz-automation/fledge-north-iec104 into RFD-86

jude-adam committed:
Fixed pem certificate search folder (RFD-82)

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

Commit:
Added testing to data on information objects

Signed-off-by: Maurício <mauricio.fernandes@gridcorelabs.com>

mzillgith committed:
- fixed test case InterrogationHandlerTest.InterrogationHandlerBroadcastCA

Signed-off-by: Michael Zillgith <michael.zillgith@mz-automation.de>

Commit:
Fixed TLSConnectionOnlyKnownCertsFalse test and replaced expired certificates

Signed-off-by: Maurício <mauricio.fernandes@gridcorelabs.com>

jude-adam committed:
Merge branch 'RFD-86' of bitbucket.org:mz-automation/fledge-north-iec104 into main

jude-adam committed:
Changed Step Position Value to new string format

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

aklira committed:
Merge pull request #42 from fledge-power/mz_refactoring

Mz refactoring

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

jude-adam committed:
Fixed issue with step position information and also added tests

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

jude-adam committed:
Fixed tests

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

Commit:
refs #43 Added utility log functions to have all logs displayed in stdout during unit tests. Removed unused printf. Includes cleanup.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #43 Added prefix to all existing logs. Added unit test for new log utility functions. Added CMake option for logs in unit tests.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
refs #44 - Container HNZ

Signed-off-by: Yannick Marchetaux <yannick.marchetaux@equans.com>

YmaIneo committed:
refs #44 - Correction path

Signed-off-by: Yannick Marchetaux <yannick.marchetaux@equans.com>

Commit:
refs #43 Refactored part of the code to add more relevant logs. Fixed some member variables missing initialization. Added try/catch to protect conversions from string to number variables. Fixed a potential crash in case redundancy group had no rg_name.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
refs #44 - Correction container name

Signed-off-by: Yannick Marchetaux <yannick.marchetaux@equans.com>

YmaIneo committed:
Merge pull request #46 from fledge-power/43-logs-enhancement

43 logs enhancement

YmaIneo committed:
Merge pull request #45 from fledge-power/44-Container-HNZ-YMA

refs #44 - Container HNZ

Commit:
refs #43 Added extra verification + formatting in IEC104Config.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #48 from fledge-power/43-logs-enhancement

refs #43 Added extra verification + formatting in IEC104Config.

Commit:
refs #49 Workaround for Fledge operation always returning -1.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #50 from fledge-power/49-all-operations-are-considred-failed

refs #49 Workaround for Fledge operation always returning -1.

jude-adam committed:
Moved slave starting to plugin_start instead of plugin_init

Signed-off-by: Jude Mingay <jude.mingay@gridcorelabs.com>

aklira committed:
Merge pull request #54 from fledge-power/mz_refactoring

Mz refactoring

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
update to JDK 17

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

Commit:
Merge branch 'main' into develop

# Conflicts:
#	src/iec104.cpp
#	tests/test_control.cpp
#	tests/test_info.cpp

aklira committed:
Merge pull request #55 from fledge-power/develop

Bring logs update from develop to main

aklira committed:
Update publish.yml

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
Update publish.yml

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
Update publish.yml

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
IEC104Server::send log uses ts in ns using Hal_getTimeInNs

IEC104Server::send log uses ts in ns using Hal_getTimeInNs

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
added audit logs for connection monitoring

Signed-off-by: aklira <akli.rahmoun@gmail.com>

aklira committed:
Update VERSION

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

aklira committed:
use of NHDWN, NHAVL audit sources instead of SRVFL

Signed-off-by: AkliRa <39710290+aklira@users.noreply.github.com>

Commit:
Fixed the log prefix for connectionEventHandler function.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

c-descoust committed:
refs #PMP2-302 Encode audit messages as a json string

Signed-off-by: Charles DESCOUST <charles.descoust@external.equans.com>

c-descoust committed:
refs #PMP2-302 Use of the SRVFL common audit type

Signed-off-by: Charles DESCOUST <charles.descoust@external.equans.com>

c-descoust committed:
refs #PMP2-302 Apply same logic than south-iec104 for audits

Signed-off-by: Charles DESCOUST <charles.descoust@external.equans.com>

Commit:
refs #PMP2-286 Send negative ACK for command when south is not connected. Send negative ack for command when rejected with COT 44 or 47.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-286 Fixed crash in unit test when IEC104 server could not be created but monitoring thread attempted to start it anyway.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-286 Added missing negative flag to some ACK messages. Added missing ACT-CON COT for an ACK message.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Ineo-JFA committed:
refs #PMP2-302 : Audits from lib60870 events

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-302 : Unit test for audits

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Commit:
refs #PMP2-384 Updatec CI pipeline to use the correct version of the lib. Added extra checks to unit tests.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-384 Commented tests related to TLS that fails with lib60870 v2.3.3.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-384 Fixed env variable not accessible in CI.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-384 Fixed sonar scanner version.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #62 from fledge-power/PMP2-384-freeze-versions-iec-libs

Pmp2 384 freeze versions iec libs

Ineo-JFA committed:
refs #PMP2-302 : Fix log

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-357 : Fix timestamp quality flags

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-357 : Unit test timestamp quality flags

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-357 : Sonarqube quality fix

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Commit:
refs #PMP2-268 Added and updated unit tests related to commands.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Ineo-JFA committed:
refs PMP2-302 : Sonarqube quality adaptations

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Commit:
refs #PMP2-268 Sonarqube code quality fixes.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-268 Removed workaround as fledge-iot#1210 was fixed.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #64 from fledge-power/PMP2-357-Fix_timestamp_quality_flags

PMP2-357 : Fixed timestamp quality flags for TS

YmaIneo committed:
Merge pull request #63 from fledge-power/PMP2-302_Harmonize_audit_categories

PMP2-302 : New audit categories

Commit:
refs #PMP2-286 Code review fixes.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-286 Sonarqube code quality fixes part 2.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #65 from fledge-power/PMP2-286-manage-negative-acks

Pmp2 286 manage negative acks

Commit:
refs #PMP2-302 Fixed global disconnect audit being sent after losing connection on only one path.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

Commit:
refs #PMP2-302 Attempt to fix CI not running due to cache plugin version being too old.

Signed-off-by: Florent Peyrusse <florent.peyrusse@external.equans.com>

YmaIneo committed:
Merge pull request #66 from fledge-power/PMP2-302_Fix_global_audits

refs #PMP2-302 Fixed global disconnect audit being sent after losing …

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #67 from fledge-power/develop

Develop

Ineo-JFA committed:
refs #PMP2-490 : Fix audit details format

Signed-off-by: FERSULA Jeremy <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-486 : Add coverage exclusions

Unreachable branches are generated from the CFG, leading to misleading coverage statistics.
Policy for exclusion in this commit is as follow :

* Logs
* string declaration for logs headers
* Mutex lock
* break keywords

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-486 : Exclude multiline logs from coverage

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #68 from fledge-power/PMP2-490-Fix_audit_format

refs #PMP2-490 : Fix audit details format

marie-miceli committed:
refs #PMP2-390 Upgrade certificates from sha1 to sha256

Signed-off-by: Marie Miceli <marie.miceli@external.equans.com>

marie-miceli committed:
refs #PMP2-390 Deactivating broken tests in connectionHandler

Signed-off-by: Marie Miceli <marie.miceli@external.equans.com>

marie-miceli committed:
refs #PMP2-390 Line from commented test forgotten

Signed-off-by: Marie Miceli <marie.miceli@external.equans.com>

YmaIneo committed:
Merge pull request #69 from fledge-power/PMP2-390-Updating_SHA1_certificates

PMP-2 refs #390 Updating sha1 certificates

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #70 from fledge-power/develop

Merge for release 1.2.4

Ineo-JFA committed:
refs PMP2-521 : Add operations to sync init with south

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-521 : Add unit test SyncConnectionWithSouth

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #72 from fledge-power/PMP2-521-Ordered_initial_connexion

refs PMP2-521: Ordered initial connexion

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #73 from fledge-power/develop

Develop PMC-2.2.2

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

niels-boussion committed:
refs #PMP2-297 Set ci version to ubuntu-22.04

Signed-off-by: Niels BOUSSION <niels.boussion@equans.com>

niels-boussion committed:
refs #PMP2-297 Updated ci ubuntu to 24.04

Signed-off-by: Niels BOUSSION <niels.boussion@equans.com>

Ineo-JFA committed:
refs PMP2-297: Force C++11 in coverage mode

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-297: Rewrite ambiguous casts to char*

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs #PMP2-486 : Exclude additionnal strings from coverage

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Fix unit test IP

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Exclude return on switch case from coverage

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Add tests ValidateCommands and SendCommands

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Improve coverage

* Rewrite test SendCommands
* Exclude mutex locks from coverage

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Remove duplicated test PluginInfo

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Disable optimization for coverage

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
refs PMP2-486 : Fix char* comparison

Signed-off-by: Jérémy FERSULA <jeremy.fersula@external.equans.com>

Ineo-JFA committed:
Update CMakeLists.txt

Add spacing to CMAKE_C_FLAGS

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #74 from fledge-power/PMP2-486-Test_coverage_adaptations

Overall tests adaptations

Ineo-JFA committed:
Update VERSION

Signed-off-by: Jérémy Fersula <jeremy.fersula@external.equans.com>

YmaIneo committed:
Merge pull request #75 from fledge-power/develop

Merge for release v2.0.0
